components/user-store/org.wso2.carbon.identity.user.store.count/src/main/java/org/wso2/carbon/identity/user/store/count/UserStoreCountService.java

```diff
@@ -202,4 +173,58 @@ public Long countByClaimInDomain(String claimURI, String valueFilter, String dom
 
     }
 
+    private RealmConfiguration realmConfiguration = null;
+
+    /**
+     * Get User count.
+     *
+     * @param claimURI claim uri
+     * @param valueFilter filter
+     * @return user count
+     * @throws UserStoreCounterException UserStoreCounterException
+     */
+    private long getUserCountWithClaims(String claimURI, String valueFilter) throws UserStoreCounterException {
+
+        try {
+            realmConfiguration = CarbonContext.getThreadLocalCarbonContext().getUserRealm().getRealmConfiguration();
+            return UserStoreCountDSComponent.getRealmService().getUserRealm(this.realmConfiguration).
+                    getUserStoreManager().getUserCountWithClaims(claimURI, valueFilter);
+
+        } catch (UserStoreException e) {
+            String ErrorMsg = "No user count is retrieved from the user store.";
+            throw new UserStoreCounterException(ErrorMsg, e);
+        }
+    }
+
+    /**
+     * Get role count.
+     *
+     * @param filter filter
+     * @return role count
+     * @throws UserStoreCounterException UserStoreCounterException
+     */
+    private long getCountRoles(String filter) throws UserStoreCounterException {
+
+        try {
+            realmConfiguration = CarbonContext.getThreadLocalCarbonContext().getUserRealm().getRealmConfiguration();
+            return UserStoreCountDSComponent.getRealmService().getUserRealm(this.realmConfiguration).
+                    getUserStoreManager().countRoles(filter);
```

Comment:
comments added for above `getUserCountWithClaims` needs to be fixed for this method as well.